react-component-template/index.jsx

```diff
@@ -12,19 +14,8 @@ export default class {{PascalName}} extends Component {
     name: '{{titleName}}'
   }
 
-  constructor (props) {
-    super(props)
-    // initialize state values
-    this.state = {}
-  }
-
-  // use the pure-render mixin without the mixin. This allows us to use es6
-  // classes and avoid "magic" code. NOTE: if this component is used directly
-  // by react-router, you should delete it, otherwise, the <Link> component will
-  // not cause a re-render
-  shouldComponentUpdate (...args) {
-    return shouldComponentUpdate.apply(this, args)
-  }
+  // initialize state values
+  state = {}
```

Comment:
The original idea to initialize state in the constructor everytime was to make it clear to users where state should/could be initialized. Just a helpful template really. If this causes any problems, such as hindering development, making the code difficult to read, or slowing down our app, then I would fully support removing it. Otherwise, I don't feel strongly about making this change. It seems like a personal preference issue, and I can work with either solution.